src/wpool/mongoose_wpool_rdbms.erl

```diff
@@ -0,0 +1,54 @@
+-module(mongoose_wpool_rdbms).
+-behaviour(mongoose_wpool).
+
+-export([init/0]).
+-export([start/4]).
+-export([stop/2]).
+
+init() ->
+    case ets:info(prepared_statements) of
+        undefined ->
+            Heir = case whereis(ejabberd_sup) of
+                       undefined -> [];
+                       Pid -> [{heir, Pid, undefined}]
+                   end,
+            ets:new(prepared_statements,
+                    [named_table, public, {read_concurrency, true} | Heir]),
+            ok;
+        _ ->
+            ok
+    end.
+
+start(Host, Tag, WpoolOpts, RdbmsOpts) ->
+    try do_start(Host, Tag, WpoolOpts, RdbmsOpts)
+    catch
+        Err -> {error, Err}
+    end.
+
+do_start(Host, Tag, WpoolOpts0, RdbmsOpts) ->
+    Backend =
+        case lists:keyfind(server, 1, RdbmsOpts) of
```

Comment:
I don't know if it's only me, but I really don't like when the result of a case clause is assigned to a var...

Reply:
Any comment on this?